###### TeleCall uses 4 centers around the globe to process customer order forms. They audit a certain %  of the customer order forms. Any error in order form renders it defective and has to be reworked before processing.  The manager wants to check whether the defective %  varies by centre. Please analyze the data at 5% significance level and help the manager draw appropriate inferences

In [1]:
import pandas as pd

In [2]:
data = pd.read_csv('Costomer+OrderForm.csv')
data

Phillippines   Indonesia       Malta       India
0     Error Free  Error Free   Defective  Error Free
1     Error Free  Error Free  Error Free   Defective
2     Error Free   Defective   Defective  Error Free
3     Error Free  Error Free  Error Free  Error Free
4     Error Free  Error Free   Defective  Error Free
..           ...         ...         ...         ...
295   Error Free  Error Free  Error Free  Error Free
296   Error Free  Error Free  Error Free  Error Free
297   Error Free  Error Free   Defective  Error Free
298   Error Free  Error Free  Error Free  Error Free
299   Error Free   Defective   Defective  Error Free

[300 rows x 4 columns]

In [3]:
phillippiness_indonesia_data = pd.crosstab(index = data.Phillippines, columns = data.Indonesia, margins=True)
phillippiness_indonesia_data

Indonesia     Defective  Error Free  All
Phillippines                            
Defective             2          27   29
Error Free           31         240  271
All                  33         267  300

In [4]:
phillippiness_malta_data = pd.crosstab(index = data.Phillippines, columns = data.Malta, margins=True)
phillippiness_indonesia_data

Indonesia     Defective  Error Free  All
Phillippines                            
Defective             2          27   29
Error Free           31         240  271
All                  33         267  300

In [5]:
Malta_India_data = pd.crosstab(index = data.Malta, columns = data.India, margins=True)
Malta_India_data

India       Defective  Error Free  All
Malta                                 
Defective           0          31   31
Error Free         20         249  269
All                20         280  300

In [6]:
import scipy.stats as stats
chi_square_tscore,pval,df,exp_table = stats.chi2_contingency(phillippiness_indonesia_data)
print('Chi-square score is {}.\np value is {}.\ndf is {}.\nExpected table is \n{}'.format(chi_square_tscore,pval,df,exp_table))

Chi-square score is 0.5521603241815238.
p value is 0.9682285093497439.
df is 4.
Expected table is 
[[  3.19  25.81  29.  ]
 [ 29.81 241.19 271.  ]
 [ 33.   267.   300.  ]]


In [7]:
import scipy.stats as stats
chi_square_tscore,pval,df,exp_table = stats.chi2_contingency(Malta_India_data)
print('Chi-square score is {}.\np value is {}.\ndf is {}.\nExpected table is \n{}'.format(chi_square_tscore,pval,df,exp_table))

Chi-square score is 2.469463621879979.
p value is 0.650112024745744.
df is 4.
Expected table is 
[[  2.06666667  28.93333333  31.        ]
 [ 17.93333333 251.06666667 269.        ]
 [ 20.         280.         300.        ]]


###### The p_value = 0.65011 > 0.05 so accept null hypothesis
inferences: as defective percentage is same in all countries there is no need of any action .